js/src/main/scala/JSVyxal.scala

```diff
@@ -1,42 +1,48 @@
 package vyxal
 
-import org.scalajs.dom
-import org.scalajs.dom.document
+import scala.annotation.static
 import scala.scalajs.js
 import scala.scalajs.js.annotation.{JSExport, JSExportTopLevel}
 import scala.scalajs.js.JSConverters.*
 
 /** A bridge between the interpreter and JS */
 @JSExportTopLevel("Vyxal")
 object JSVyxal:
+  val output: StringBuilder = StringBuilder()
   @JSExport
-  def execute(code: String, inputs: String, flags: String): String =
+  def execute(
+      code: String,
+      inputs: String,
+      flags: String,
+      printFunc: String
+  ): Unit =
     // todo take flags to set settings
     // todo take functions to print to custom stdout and stderr
-    val output = document.getElementById("output")
-    // cast output object to textarea
-    val textarea = output.asInstanceOf[dom.html.TextArea]
-    textarea.value = ""
-    val settings =
-      Settings(online = true)
+    val settings = flags.foldLeft(Settings(online = true))(_.withFlag(_))
     val globals = Globals(
-      settings = settings
+      settings = settings,
+      printFn = (x: VAny) =>
```

Comment:
This cursed conglomerate of a print function is so that programs can live stream their input to the output box and still be interrupted. It's not just a neat feature, it's a consequence of basically the only way I could think of allowing programs to be stopped.